runtime/runtime/src/ext.rs

```diff
@@ -35,6 +35,18 @@ pub struct RuntimeExt<'a> {
     last_block_hash: &'a CryptoHash,
     epoch_info_provider: &'a dyn EpochInfoProvider,
     current_protocol_version: ProtocolVersion,
+
+    #[cfg(feature = "protocol_feature_function_call_ratio")]
+    distribute_leftover_gas_to: Vec<GasRatioMetadata>,
+    #[cfg(feature = "protocol_feature_function_call_ratio")]
+    gas_ratio_sum: u64,
+}
+
+#[cfg(feature = "protocol_feature_function_call_ratio")]
+struct GasRatioMetadata {
```

Comment:
We try to avoid `u*` types in the runtime as much as possible. If you find usage of `u*` type anywhere please file a Github issue.

Reply:
I must say I don't entirely understand what should be used instead. As this interfaces with `VmLogic` which used raw `u64/u32`, we actually have a whole bunch of `u64`s in various signatures in this module. 

But yeah, I'd probably spell this as:

```rust
struct RuntimeExt {
  ...
  gas_weights: Vec<(FunctionCallActionIndex, GasWeight)>
}
struct GasWeight(u64);

struct FunctionCallActionIndex {
  receipt_index: usize,
  action_inde
}

impl RuntimeExt {
    fn get_function_call_action_mut(&mut self, index: FunctionCallActionIndex) -> &mut FunctionCallAction {
    }
}
```

* Separate the two-dimensional index into its own type
* Add a typed wrapper to the weight
* Remove `gas_ratio_sum` field -- it seems we can compute it in the `distribute` function. To check if it's zero, you can check if `gas_weights` is not empty

Reply:
Do you want the u64 removed from the trait interface or just internally in the implementation? I notice for the other types in the signature the types are aliases in the trait definition but then just use the primitive type in the implementation.

As for the proposed API, it wouldn't work with the current implementation because the gas weights need to be borrowed mutably so I can't take a mutable reference on `self` when it's used. Seems more complex to read with this change to me, but happy to change to whatever is preferred.

Reply:
So I've changed to just wrapping internally for now, but can expose this type through the trait interface if that's what we want to move the API toward